--- job.py	2018-03-02 18:05:10.004015578 +0000
+++ job_new.py	2018-03-03 06:38:27.481891364 +0000
@@ -387,6 +387,11 @@
 
         runtime = [u"docker", u"run", u"-i"]
 
+        # add the volume for the docker socket on the host so we can create containers on the host
+        runtime.append(u"--volume=%s:%s:rw" % ("/var/run/docker.sock", "/var/run/docker.sock"))
+        # add a volume for the place on the host where all the containers will put data
+        runtime.append(u"--volume=%s:%s:rw" % ('/datastore', '/datastore'))
+
         runtime.append(u"--volume=%s:%s:rw" % (docker_windows_path_adjust(os.path.realpath(self.outdir)), self.builder.outdir))
         runtime.append(u"--volume=%s:%s:rw" % (docker_windows_path_adjust(os.path.realpath(self.tmpdir)), "/tmp"))
 
