--- process.py	2017-05-11 14:49:28.000000000 -0700
+++ process_new.py	2017-05-11 14:41:49.000000000 -0700
@@ -211,7 +211,7 @@
 
     def moveIt(src, dst):
         if action == "move":
-            for a in output_dirs:
+            for a in set(['/datastore']):
                 if src.startswith(a):
                     _logger.debug("Moving %s to %s", src, dst)
                     shutil.move(src, dst)
